dashboard/app/models/unit_group.rb

```diff
@@ -164,12 +164,22 @@ def write_serialization
   def update_scripts(new_scripts, alternate_scripts = nil)
     alternate_scripts ||= []
     new_scripts = new_scripts.reject(&:empty?)
+    new_scripts_objects = new_scripts.map {|s| Script.find_by_name!(s)}
     # we want to delete existing unit group units that aren't in our new list
-    scripts_to_delete = default_unit_group_units.map(&:script).map(&:name) - new_scripts
-    scripts_to_delete -= alternate_scripts.map {|hash| hash['alternate_script']}
+    scripts_to_delete = default_unit_group_units.map(&:script) - new_scripts_objects
+    scripts_to_delete -= alternate_scripts.map {|hash| Script.find_by_name!(hash['alternate_script'])}
 
-    new_scripts.each_with_index do |script_name, index|
-      script = Script.find_by_name!(script_name)
+    if scripts_to_delete.any?(&:prevent_course_version_change?)
+      raise 'Cannot remove scripts that have resources or vocabulary'
+    end
```

Comment:
I am trying to wrap my head around what we will do in the case where a curriculum writer decides that they do not want a unit in their unit group any more. perhaps the curriculum writer will have to keep the script in the unit group until they are ready to destroy it entirely? 

I don't necessarily see a better way, and I'd rather have this protection than not. SO I am ok with waiting for someone to run into this before dealing with it.

Reply:
If it doesn't have resources or vocabulary, it should be possible to remove it still. If it does, the options are going to be to go through each lesson and remove these objects or wait until they want to destroy it entirely I guess.

Reply:
I will admit the error handling is not pretty on this but I'd be surprised if this is an error path hit commonly.